src/index.ts

```diff
@@ -165,11 +169,116 @@ const extension: JupyterFrontEndPlugin<void> = {
 
     // Capture the signal
     eventEmitter.on('dataprocConfigChange', (message: string) => {
+      checkAllApisEnabled();
       if (bqFeature.enable_bigquery_integration) {
         loadBigQueryWidget('');
       }
     });
 
+    const checkAllApisEnabled = async () => {
+      const dataprocClusterResponse =
+        await RunTimeSerive.listClustersDataprocAPIService();
+
+      let bigqueryDatasetsResponse;
+      const credentials = await authApi();
+      if (credentials && credentials?.project_id) {
+        bigqueryDatasetsResponse =
+          await BigQueryService.listBigQueryDatasetsAPIService(
+            credentials?.project_id
+          );
+      }
+
+      const composerListResponse =
+        await SchedulerService.listComposersAPICheckService();
+      const dataCatalogResponse =
+        await BigQueryService.getBigQuerySearchCatalogAPIService();
+
+      if (
```

Comment:
These four categories of missing APIs seem awfully similar altogether. Do you mind drying them up a bit here for better readability? Thanks.

Reply:
Resolved